### PR TITLE
To-Do-List project for review 

### PR DESCRIPTION
**Pull request by HtetWaiYan7191**
### Requesting for a code review 

Code Review Request: JavaScript Best Practices

Code Organization: All main project files have been moved to the src folder. This ensures a more organized and structured project layout.

- Entry Point: The index.js file now serves as the main source file for all JavaScript files. It acts as the entry point for the application.

- Module Organization: All JavaScript functions and modules are now located within the modules folder. This enhances code maintainability and separation of concerns.

- Webpack Integration: The project now utilizes Webpack for bundling the JavaScript code. This enables better dependency management and optimizes the production build.

- Linter Implementation: We have introduced linters to ensure clean code and adherence to best practices. These linters help identify and fix potential issues early in the development process.

### CODE REVIEW

**Review by olanikegloria — CHANGES_REQUESTED** [review]
Please separate the code into smaller, reusable functions and modules. This improves readability and maintainability.
Instead of adding event listeners to each individual task, you can use event delegation to listen for events on the parent container (.tasks-container). This way, you can handle events for dynamically added tasks without attaching multiple listeners.
const loadDom = () => { document.addEventListener('DOMContentLoaded', showTask); }; loadDom(); should replace this function to
function handleDOMContentLoaded() {
  showTask();
}

// Event listeners
document.addEventListener('DOMContentLoaded', handleDOMContentLoaded);

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  storeLocalStorage(storeTasks);
+  showTask();
+});
+
+const loadDom = () => {
+  document.addEventListener('DOMContentLoaded', showTask);
+};
+loadDom();
+document.addEventListener('DOMContentLoaded', showTask);
+addBtn.addEventListener('click', addTask);
```
Comment (olanikegloria) [inline src/index.js]: const loadDom = () => { document.addEventListener('DOMContentLoaded', showTask); }; loadDom(); should replace this function to
function handleDOMContentLoaded() {
  showTask();
}

// Event listeners
document.addEventListener('DOMContentLoaded', handleDOMContentLoaded);

**Review by olanikegloria — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle:
hello , @HtetWaiYan7191 
Good job so far !
 Tasks have required attributes However, there are some issues that you still need to work on to go to the next project but you are almost there ! :+1:
:+1: No Linters issue
:+1: Good PR
:+1: Good use of Javascript
## Required Changes
Kindly consider to using event delegation for your load Dom function. Doing that will make your code more clear and follows best practices. :+1:
Cheers and Happy Coding ! :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100 % clear. And please, remember to tag me in your question so I can receive the notification

File: src/index.js (hunk 1 above)
Comment (HtetWaiYan7191) [reply src/index.js]: Thank you very much for explaining in details @olanikegloria 😁 . I am going to replace my loadDom function as you said 💯 .

**Review by olanikegloria — APPROVED** [review]
Status: APPROVED 🟢
Hi [@HtetWaiYan7191 ](https://github.com/HtetWaiYan7191) 👋 👋
I'm [@olanikegloria ](https://github.com/olanikegloria), Your code reviewer,
Good job on implementing the project requirements! 🥇

**Pull request merged**